Move Fatiando sample data to Rockhound

santisoler opened the pull request:
Add functions to fetch sample datasets from Harmonica and Verde.
Add South Africa gravity and British aeromagnetic anomalies datasets.
Add gallery examples for new datasets.
Set `dask`, `rasterio` and `netcdf4` as optional dependencies.

After last [Fatiando Community Call](https://github.com/fatiando/meeting-notes/blob/master/community-calls/2020.md#2020-06-19) we decided to move all sample data from Harmonica and Verde to Rockhound.


**Reminders**:

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst` and the base `__init__.py` file for the package.
- [ ] Write detailed docstrings for all functions/classes/methods. It often helps to design better code if you write the docstrings first.
- [ ] If adding new functionality, add an example to the docstring, gallery, and/or tutorials.
- [ ] If adding new example, please build the docs with `make -C doc gallery` and push the changes made on `doc/gallery`. Continuous Integration services build and deploy the docs, but they won't run the examples.
- [ ] Add your full name, affiliation, and [ORCID](https://orcid.org) (optional) to the `AUTHORS.md` file (if you haven't already) in case you'd like to be listed as an author on the [Zenodo](https://zenodo.org/communities/fatiando) archive of the next release.


**Reminders for Maintainers**:

- [ ] Maintainers **must** run all tests locally before the PR is merged. CIs run only minimal tests and styling checks.

santisoler committed:
Start moving Fatiando sample data to Rockhound

leouieda commented on rockhound/sample_data.py:
```diff
@@ -0,0 +1,201 @@
+        The bathymetry data. Columns are longitude, latitude, and bathymetry
+        (in meters) for each data point.
+
+    See also
+    --------
+    setup_baja_bathymetry_map: Utility function to help setup a Cartopy map.
```
These should probably be removed since we're not porting the plotting functions here.

santisoler committed:
Set dask, netcdf4 and rasterio as optional dependencies

santisoler committed:
Remove see also section on docstrings

santisoler committed:
Add missing reference to Forste 2014

santisoler committed:
Add verde as an optional dependency

santisoler committed:
Add examples for new datasets

Also move the South Africa gravity and British magnetic anomaly to the
datasets category, removing them from the sample datasets.
Add new functions to API references.

santisoler committed:
Add test functions for new datasets

santisoler committed:
Remove plotting functions

The code for improving Cartopy maps has been moved to the gallery
examples.

santisoler committed:
Remove unused imports

santisoler commented:
@leouieda I think this is ready. I just need to upload the gallery files, but I'll do it just before merging so the review doesn't have too many files.

Remember that the CIs are just running style checks and availability tests. Please run the full tests and compile the gallery before approving the review.

Just a quick note. I'm experiencing the same problem as Verde (https://github.com/fatiando/verde/issues/269): Cartopy is not correctly plotting lands and/or oceans. I tried to pin cartopy to `0.17.*` but got a `Connection timed out` error when cartopy was trying to download http://naciscdn.org/naturalearth/110m/physical/ne_110m_land.zip . Probably the website is currently down and I should try again later. If it end up working just fine, I will pin it and push the new `environment.yml` and `requirements-dev.txt`.

leouieda commented on doc/api/index.rst:
```diff
@@ -18,6 +18,21 @@ Data Fetching and Loading
+    fetch_south_africa_gravity
+
+Sample datasets
+---------------
```
```suggestion
---------------

These are small datasets (or subsets of larger data) that are used to test methods
and in project documentation and tutorials. 
```

leouieda commented on environment.yml:
```diff
@@ -8,10 +8,11 @@ dependencies:
     - xarray
     - pandas
+    # Development requirements
+    - verde
```
Do we really need to pull in `verde`? It seems like it's just for the `maxabs` function, which we could do without.

santisoler replied:
You're right, we can ditch Verde as a dependency and compute the maxabs trough Numpy.

leouieda commented:
Trying to build the gallery locally but running into the same issue with the Cartopy download. Will try again later on.

santisoler committed:
Add sample datasets description

Co-authored-by: Leonardo Uieda <leouieda@gmail.com>

santisoler committed:
Ditch Verde as an optional dependency

Verde was only being used by some gallery examples for getting regions
and max absolute values. We can safely ditch Verde and use numpy
instead.

leouieda commented:
I managed to test the gallery locally finally. I'm making some edits and will push them soon. I also checked that we can download individual files from a Zenodo archive.

leouieda committed:
Move sample data examples to a sub gallery

This introduces a separation in the gallery between the actual
models/data and the sample data.

leouieda reviewed (approved):
@santisoler this looks good to me :+1: I moved the sample data plots to a sub gallery so that they appear separately. Otherwise, this is good to merge!

santisoler commented:
Thanks @leouieda!

> I also checked that we can download individual files from a Zenodo archive.

Great! We should upload all sample data files to Zenodo and then change the URLs in the registry file.
@mtb-za would you like to tackle it? I think we might need to add you to the Fatiando group in Zenodo to do so, but after that you'll be ready to start uploading the files.

mtb-za commented:
I will carve out some time for it this week, yeah.

mtb-za commented:
What sort of license can we use?
Is everything fine with CC-0 (because it is public domain), or do we need something else?

For now, I am going to call this a version `1.0.0` upload, containing everything in https://github.com/fatiando/verde/tree/master/data https://github.com/fatiando/harmonica/tree/master/data including a combined readme. Am I missing a data file somewhere? (Boule, Pooch and Rockhound do not have any.)

I am also not too sure what to do for authors, since none of us are really the authors of any of these datasets. Can we credit the agencies as authors directly?

Pinging @leouieda @santisoler.

santisoler commented:
> What sort of license can we use? Is everything fine with CC-0 (because it is public domain), or do we need something else?

This is tricky, because not every dataset has the same license. For example, the British aeromagnetic data is released under the [Open Goverment License](http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/) which requires acknowledgement, so we cannot publish it under public domain. Don't know if we can omit a global license for the whole upload and specify the license of each dataset on a README.

> For now, I am going to call this a version 1.0.0 upload, containing everything in https://github.com/fatiando/verde/tree/master/data https://github.com/fatiando/harmonica/tree/master/data including a combined readme. Am I missing a data file somewhere? (Boule, Pooch and Rockhound do not have any.)

Yes, we only need to upload datasets from Verde and Harmonica. Although, don't add the Rio magnetic data: it doesn't have a clear license and we will deprecate it on Verde 2.0.0. Users will still be able to download it until that release (for keeping backward compatibility), but they are encouraged to use another dataset.

> I am also not too sure what to do for authors, since none of us are really the authors of any of these datasets. Can we credit the agencies as authors directly?

Maybe we can put something as Fatiando Developer Team? What do you think @leouieda ?

mtb-za commented:
> Don't know if we can omit a global license for the whole upload and specify the license of each dataset on a README.

It looks like we can do multiple uploads with different licenses. We should therefore try and group them, if any of them have the same license. Not sure if I can add the next lot to the same DOI though, I will have to experiment. We might just have to bite the bullet and have a new upload for each file.
This is the info under the text-box for entering the license detail:
>Selected license applies to all of your files displayed on the top of the form. If you want to upload some of your files under different licenses, please do so in separate uploads. If you cannot find the license you're looking for, include a relevant LICENSE file in your record and choose one of the Other licenses available (Other (Open), Other (Attribution), etc.). The supported licenses in the list are harvested from opendefinition.org and spdx.org . If you think that a license is missing from the list, please contact us.

> Although, don't add the Rio magnetic data: it doesn't have a clear license and we will deprecate it on Verde 2.0.0. Users will still be able to download it until that release (for keeping backward compatibility), but they are encouraged to use another dataset.

I will leave the Rio data out of the upload. Do we have an alternative magnetic dataset to upload instead?

santisoler commented:
Yes, we have the British one.
